### ISSUE TITLE
Left-Handed Keybind Preset

### ISSUE THREAD

**Issue by the-phinet** [UZDoom/UZDoom#1072]
### Game

All

### Problem Description

> Funnily enough, as a left handed player, WASD really doesn't suit me at all and I actually found the move to not using the arrow keys as a default quite jarring. Even now, when I do a clean install and press the arrows keys to find them doing nothing, it still feels like something has broken. So, personally, I'd be quite happy if it wasn't the default. I get that people tend to use WASD, so I don't actually think it should be removed as a default, but it's a really awkward one for me. That's what options are for though, and the keybinds are one of the things I change immediately. So, for me WASD is not a good default, but I accept that it probably is for the majority. So, I, agree with you.
>
> Similarly (and I'm speaking more generally about all modern games - so, basically off-topic), keys near WASD being used for use, open, alt-fire, scan, crouch, hack, lock-pick, whatever are, perhaps, even more awkward - they are on the wrong side of the keyboard for me, lost somewhere in the middle of the rest of the keys, involve me taking my hand off the arrows, and are really hard to find in a tight situation. The number of games that do not allow rebinding of such keys is surprisingly high. At least full binding/rebinding of everything is possible in UZDoom.

### Feature Description

Add an optional keybind preset that is suitable for left-handed players. I'm not sure what this would look like, as I am not left-handed.

If some southpaw users could share their setups, it would be very appreciated. Examples of what commercial games offer would also be useful.

**Comment by Enjay001** [comment]
As the person who made the original comment, there is actually a preset in the game already that comes pretty close to what works for me: the "modern (keypad)" one.  (The point of my original post was that WASD doesn't suit me as a default, not that there isn't a preset that is close to what I need.)

If a left handed player uses the mouse in their left hand (as I do) then, naturally, the right hand is used for keyboard button-presses.  That means that keys on the right-hand side of the keyboard are best (and that is where the keypad and discrete arrow buttons are, of course).  So, "modern (keypad)" is already doing most of the heavy lifting.

(WASD is really awkward - they are on the wrong side of the keyboard and their offsets are such  (S being under but slightly to the right of W) that you have to twist your wrist round uncomfortably or angle the keyboard to use them.)

What I need is the up arrow to be move forward, down to be backwards and left/right to be move (not turn) left/right.  The "modern (keypad)" setting provides that - both on the keypad and on the discrete arrows.

I personally find the arrow keys easier to use than the keypad.  I suspect that's a combination of habit (Wolfenstein, Doom, ROTT, etc "training" me to use them (though I always used the keypad for Eye of the Beholder because of how turning works there)) but also that I find the clear space around the keys useful.  When I'm using them, they are the only keys in that immediate area and I won't hit anything else by mistake, but there are nearby keys that I can bind to things if I need to.   So, again, "modern (keypad)" does do most of what I need.

The only thing I don't get the advantage of is that "modern (keypad)" also sets the 7/Home key as jump and 1/End as crouch.  They are obviously nice and close to the keypad movement keys.  So the setup makes sense.  It's just that I prefer the discrete arrows (as mentioned).  The (nearly) equivalent buttons on the little pad above the arrows (home and end) don't auto-map to those functions with the "modern (keypad)" preset and the end key already has another function bound to it.  I probably wouldn't use them anyway - there are other keys nearer the arrows that I find more convenient.

My personal preferences beyond the arrows are probably not a particularly good as generalised defaults.  I use the middle mouse button for jump, the Ctrl button to the left of the arrow keys for crouch (toggle) and additional alt-fire/zoom weapon functions go on the additional buttons of my mouse.  So, like I said, I don't really think that's something that really needs or should be included as a general preset option.  Some of those preferences are less to do with being left-handed and more to do with what works specifically for me.

There may be someone else who chips in and says "for a left handed set up I need..." but I actually think that the "modern (keypad)" preset probably already comes close to what is needed and further tailoring is just part of what needs to be done when installing and setting up a game.  Perhaps to help users recognise it, it could be renamed to include left-handed in the name, or a tooltip saying something like "may be better suited to left-handed players" could be added?  It took me a while to realise that it might be useful to a lefty.

Perhaps the only consideration for "modern (keypad)" is that it does create a nice left-hand friendly set of bindings, complete with jump and crouch bound to something nearby, but not all keyboards have a keypad.  MAybe some consideration could be given to what a left-handed setup for a laptop that doesn't have a keypad (or those keys are only available after pressing some function key or other) might be like?  I'm not sure.  I have a couple of laptops like that, but I have never used them for playing.


So, certainly from my perspective, "modern (keypad)" is actually a pretty good starting point for a left-handed setup, and I only have to alter it because of personal preferences rather than left-handed needs.

Thanks for picking up the comment and acting on it though. :)

**Comment by EdHerdman** [comment]
Keypad as a default does have a drawback in that it's missing on tenkeyless keyboards. I bet there are some other holdouts like me who think it's too far away or they prefer to have their hands closer to the normal typing keys, but YMMV.

I have an old-fashioned setup somewhat evolved from DOOM / Heretic and Quake binds, with arrow keys, Enter/Return for jump, Shift for walk toggle, Caps Lock for run toggle, and the Home row is my first choice for various functions. Don't expect many people will like exactly that, though.
